tests/integration_tests/datasources/test_none.py

```diff
@@ -0,0 +1,64 @@
+"""NoCloud datasource integration tests."""
+import json
+
+import pytest
+from pycloudlib.lxd.instance import LXDInstance
+
+from tests.integration_tests.decorators import retry
+from tests.integration_tests.instances import IntegrationInstance
+from tests.integration_tests.integration_settings import PLATFORM
+from tests.integration_tests.util import verify_clean_log
+
+DS_NONE_BASE_CFG = """\
+datasource_list: [None]
+datasource:
+  None:
+    metadata:
+      instance-id: my-iid-uuid
+    userdata_raw: |
+      #cloud-config
+      runcmd:
+      - touch /var/tmp/success-with-datasource-none
+"""
+
+LXD_METADATA_DS_NONE_BASE_CFG = """\
+  /etc/cloud/cloud.cfg.d/99-force-dsnone.cfg:
+    when:
+    - create
+    - copy
+    create_only: false
+    template: dsnone.tpl
+    properties: {}
+"""
+
+
+@retry(tries=30, delay=1)
+def wait_for_cloud_init_status_file(instance: LXDInstance):
+    status_file = instance.read_from_file("/run/cloud-init/status.json")
+    assert len(status_file)
+
+
+@pytest.mark.lxd_use_exec
+@pytest.mark.skipif(
+    PLATFORM != "lxd_container",
+    reason="Requires LXD container with custom metadata setup",
+)
+def test_datasource_none_discovery(client: IntegrationInstance):
+    """Integration test for #4635.
+
+    Test that DataSourceNone detection (used by live installers) doesn't
+    generate errors or warnings.
+    """
+    log = client.read_from_file("/var/log/cloud-init.log")
+    verify_clean_log(log)
+    # Limit datasource detection to DataSourceNone.
+    client.write_to_file(
+        "/etc/cloud/cloud.cfg.d/99-force-dsnone.cfg", DS_NONE_BASE_CFG
+    )
+    client.execute("cloud-init clean --logs --reboot")
+    wait_for_cloud_init_status_file(client)
+    status = json.loads(client.execute("cloud-init status --format=json"))
+    assert [] == status["errors"]
+    assert {} == status["recoverable_errors"]
+    client.execute("cloud-init status --wait")
+    assert client.execute("test -f /var/tmp/success-with-datasource-none").ok
```

Comment:
Is it worth another `verify_clean_log` call after the reboot?

Reply:
@TheRealFalcon sure. I've adapted `verify_clean_log` to append known warnings when detected DataSourceNone. We will expect to have both "Used fallback datasource"  and occasional "Cannot call Open vSwitch" dbus race.